src/backend/joanie/core/models/accounts.py

```diff
@@ -28,32 +29,55 @@ class Address(models.Model):
     uid = models.UUIDField(
         default=uuid.uuid4, unique=True, editable=False, db_index=True
     )
-    name = models.CharField(_("name"), max_length=100)
+    title = models.CharField(_("title"), max_length=100)
     address = models.CharField(_("address"), max_length=255)
     postcode = models.CharField(_("postcode"), max_length=50)
     city = models.CharField(_("city"), max_length=255)
     country = CountryField(_("country"))
+    fullname = models.CharField(_("full name"), max_length=255)
     owner = models.ForeignKey(
         User,
         verbose_name=_("owner"),
         related_name="addresses",
         on_delete=models.CASCADE,
     )
-    main = models.BooleanField(_("main"), default=False)
+    is_main = models.BooleanField(_("main"), default=False)
 
     class Meta:
         db_table = "joanie_address"
         verbose_name = _("Address")
         verbose_name_plural = _("Addresses")
+        constraints = [
+            models.UniqueConstraint(
+                condition=models.Q(is_main=True),
+                fields=["owner"],
+                name="unique_main_address_per_user",
+            )
+        ]
 
     def __str__(self):
         return f"{self.address}, {self.postcode} {self.city}, {self.country}"
 
+    def clean(self):
+        """
+        First if this is the user's first address, we enforce is_main to True.
+        Else if we are promoting an address as main, we demote the existing main address
+        Finally prevent to demote the main address directly.
+        """
+        if self.owner.addresses.count() == 0:
```

Comment:
More efficient than counting: 
```suggestion
        if not self.owner.addresses.exists():
```